### DISCUSSION TITLE
"Write" permission for members

### DISCUSSION THREAD

**Discussion by ulgens** [django-commons/membership#93] [Processes]
Hey all,

I just wanted to ask what the plan is for providing access rights for members. Do we need to select a few packages that we want to contribute to, then share their names and get "write" permission for the given repo, or is there any other approach/flow in place?

Follow up question: What "write" permissions we want to give to the members? I guess the main branches will always be protected, and they will require pull request approvals to get code in, but I guess the overall topic requires a discussion.

**Comment by tim-schilling** [comment]
Right now it's very similar to Jazzband. Anyone can request to be a member of a project which gives you triage permissions (assign issues, label them, etc). Becoming a committer requires the project team's approval, probably the admins, but may differ from team to team. This grants you the ability to push commits to the repo.

Admins requires approval of the project t admins and gives you the ability to release packages to pypi. And admin permissions on the repo in github. 

Does that make sense? Do you feel we're overlooking something or can improve somewhere?

**Reply by ulgens** [reply]
It makes sense. I wasn't able to find this documented, but now I see that it's in https://github.com/django-commons/controls?tab=readme-ov-file#new-member-playbook It's clear what should be done if they mentioned package names during their application, but I'm still not sure what one should do to get in package teams after their membership approval? Do we expect a new issue? If so, should that be in https://github.com/django-commons/membership repo? Which issue template should be used?

**Reply by tim-schilling** [reply]
If you're referring to getting a write permission to a library, I would recommend reaching out to the existing admin/committers teams and asking if they would like help. Though you should probably have a backlog of changes to the repo already to back up your offer of help too.

Once the team self-organizes, an admin request issue can be created on the membership repo.

Given the team self-organization expectation, that whole flow is a bit undefined at the moment. The hope is that people will feel empowered to ask questions and create issues as they want to. Basically, don't be shy :grin:  (with the Django Commons admin team)
